lib/front_desk.rb

```diff
@@ -0,0 +1,120 @@
+require_relative 'date_range'
+require_relative 'reservation'
+
+module Hotel
+  class FrontDesk
+    attr_accessor :reservations
+    
+    def initialize(reservations:nil )
+      @reservations = reservations || []
+    end
+
+    #access a list of all rooms in hotel
+    def list_all
+      display = ""
+      @reservations.map do |bookings| 
+        display += "Reservation_id = #{bookings.reservation_id} Start date = #{bookings.date_range.start_date} End_date =#{bookings.date_range.end_date} Room Number = #{bookings.room_num} Block Tag = #{bookings.block_tag}, Room Rate = #{bookings.room_rate}\n"
+      end 
+      return display
+    end 
+
+    #access a list of reservations for a specific room number and date range
+    def room_reservations_and_date(room_num,date_range)
+      display = ""
+      res_room_date_range = @reservations.select {|bookings| bookings.room_num == room_num && date_range.overlap(bookings.date_range)}
+      res_room_date_range.map do |bookings|
+        display += "Reservation_id = #{bookings.reservation_id} Start date = #{bookings.date_range.start_date} End_date =#{bookings.date_range.end_date} Room Number = #{bookings.room_num}\n"
+      end 
+      return display
+    end 
+
+    # check reservations of specific date
+    def check_date_reservations(date)
+      display = ""
+      reservations_include_a_date = @reservations.select {|bookings| bookings.date_range.include_date(date)==true}
```

Comment:
You don't need to compare boolean values to `true`:

```suggestion
      reservations_include_a_date = @reservations.select {|bookings| bookings.date_range.include_date(date)}
```